jupyterlab_nvdashboard/apps/gpu.py

```diff
@@ -86,73 +93,135 @@ def cb():
 
 def pci(doc):
 
-    tx_fig = figure(title="TX Bytes [MB/s]",
-                    sizing_mode="stretch_both", y_range=[0, 5000])
-    pci_tx = [pynvml.nvmlDeviceGetPcieThroughput(
-        gpu_handles[i], pynvml.NVML_PCIE_UTIL_TX_BYTES)/1024 for i in range(ngpus)]
-    left = list(range(len(pci_tx)))
+    # Use device-0 to get "upper bound"
+    pci_gen = pynvml.nvmlDeviceGetMaxPcieLinkGeneration(gpu_handles[0])
+    pci_width = pynvml.nvmlDeviceGetMaxPcieLinkWidth(gpu_handles[0])
+    pci_bw = {
+        # PCIe-Generation: (BW-per-lane / Width / 2-directions)
+        # [TODO: The specific numbers need to be validated]
```

Comment:
Do you have a recommendation on how we can validate these?

Reply:
Good question - For now, I made a change to use the specific PCIe specs listed [on wikipedia](https://en.wikipedia.org/wiki/PCI_Express).  According to those numbers, we should have max trhoughput of 15.75 GB/s in each direction for a 16-lane bus with version 3.

Reply:
Awesome!